apps/dashboard/src/components/conditions-editor/variable-select.tsx

```diff
@@ -0,0 +1,265 @@
+            setHoveredOptionIndex(0);
+            scrollToOption(0);
+          }}
+          onKeyDown={(e) => {
+            if (e.key === 'ArrowDown') {
+              next();
+              e.preventDefault();
+            } else if (e.key === 'ArrowUp') {
+              prev();
+              e.preventDefault();
+            } else if (e.key === 'Enter') {
+              if (hoveredOptionIndex !== -1) {
+                onSelect(options[hoveredOptionIndex].value ?? '');
+                setHoveredOptionIndex(-1);
+              }
+            }
+          }}
+        />
+        <header className="flex items-center justify-between gap-1 border-b border-neutral-100 bg-neutral-50 p-1">
+          <span className="text-foreground-400 text-paragraph-2xs uppercase">Variables</span>
+          <KeyboardItem>{`{`}</KeyboardItem>
+        </header>
+        <ul
+          ref={listRef}
+          // relative is to set offset parent and is important to make the scroll and navigation work
+          className="relative flex max-h-[200px] flex-col gap-0.5 overflow-y-auto overflow-x-hidden p-1"
+        >
+          {options.map((option, index) => (
+            <li
+              className={cn(
+                'hover:bg-accent text-paragraph-xs flex cursor-pointer items-center gap-1 rounded-sm p-1',
+                hoveredOptionIndex === index ? 'bg-accent' : ''
+              )}
+              key={option.value}
+              value={option.value}
+              onClick={() => {
+                onSelect(option.value ?? '');
+              }}
+            >
+              <Code2 className="text-feature size-3 min-w-3" />
+              {option.label}
+              <CheckIcon
+                className={cn('ml-auto size-4', selectedValue === option.value ? 'opacity-50' : 'opacity-0')}
+              />
+            </li>
+          ))}
+        </ul>
+        <footer className="flex items-center gap-1 border-t border-neutral-100 p-1">
+          <div className="flex w-full items-center gap-0.5">
+            <KeyboardItem>↑</KeyboardItem>
+            <KeyboardItem>↓</KeyboardItem>
+            <span className="text-foreground-600 text-paragraph-xs ml-0.5">Navigate</span>
+            <KeyboardItem className="ml-auto">↵</KeyboardItem>
+          </div>
+        </footer>
+      </div>
+    );
+  }
+);
+
+type VariableSelectProps = {
+  disabled?: boolean;
+  value?: string;
+  options: Array<{ label: string; value: string }>;
+  onChange: (value: string) => void;
+  leftIcon?: React.ReactNode;
+};
+
+/**
+ * A searchable dropdown component for selecting variables with keyboard navigation support.
+ *
+ * Features:
+ * - Filterable options list
+ * - Keyboard navigation (↑/↓ arrows)
+ * - Auto-creation of new options when typing custom values
+ * - Visual feedback for selected items
+ * - Support for custom left icon
+ */
+export const VariableSelect = ({ disabled, value, options: optionsProp, onChange, leftIcon }: VariableSelectProps) => {
```

Comment:
I hope that the above description describes it well.
![Screenshot 2025-01-14 at 12 38 07](https://github.com/user-attachments/assets/c1210ef7-c016-40f8-b244-6fac8ab50967)